Maintenance upgrades (#4341)
owncloud/docs: modules/admin_manual/pages/maintenance/upgrading/manual_upgrade.adoc

= Manual ownCloud Upgrade
:toc: right
:toclevels: 1
:page-aliases: maintenance/manual_upgrade.adoc

== Introduction

> This document describes how to manually upgrade your ownCloud installation. Post preparing the upgrade, you can decide between two paths upgrading your instance:

Reviewer: ```suggestion
This document describes how to manually upgrade your ownCloud installation. After preparing the upgrade, you can decide between two ways of upgrading your instance:
```

> .Script Guided Upgrade

Reviewer: ```suggestion
.Script-Guided Upgrade
```

This upgrade automates most of the tasks to be done including setting the correct ownership and permissions.

.Manual Step-by-Step Upgrade 

> Using this type of upgrade, you have to do all the step manually but can handle special setups.

Reviewer: ```suggestion
Using this type of upgrade, you have to do all the steps manually but you can also handle special setups.
```

NOTE: In this description we assume that your ownCloud installation was located in the default directory: `/var/www/owncloud` and the new release will reside there as well. 
The path might differ, depending on your installation.

== General Preparation

> There are several steps necessary before you can start with upgrading your owncloud instance.

Reviewer: ```suggestion
There are several steps necessary before you can start with upgrading your ownCloud instance.
```

sudo wget https://download.owncloud.org/community/owncloud-{latest-server-download-version}.tar.bz2
----

> == Script Guided Upgrade

Reviewer: ```suggestion
== Script-Guided Upgrade
```

> When using the script guided upgrade, the script from the xref:installation/manual_installation/manual_installation.adoc#script-guided-installation[Script-Guided Installation] is used from installing ownCloud. The scripts xref:installation/manual_installation/script_guided_install.adoc#script-questions[asks questions] and beside other parameters, the *upgrade an existing installation* is selected.

Reviewer: ```suggestion
When using the script-guided upgrade, the script from the xref:installation/manual_installation/manual_installation.adoc#script-guided-installation[Script-Guided Installation] is used. The script xref:installation/manual_installation/script_guided_install.adoc#script-questions[asks questions] and beside other parameters, the *upgrade an existing installation* is selected.
```

> Follow the script documentation for details installing and using it.

Reviewer: ```suggestion
Follow the script documentation for details on how to install and use it.
```

> NOTE: The script is most convenient if you have your `apps`, `apps-external` and your `data` directory linked, as it takes care of re-linking - you will get asked about when using. When having fixed directories, these are created, but content must be moved/copied manually before finalizing the upgrade. When not using the `apps-external` directory, you must manually take care copying only those apps which are not part of the new source.

Reviewer: ```suggestion
NOTE: The script is most convenient if you use links for your `apps`, `apps-external` and your `data` directory, as it takes care of recreating the links. You will be asked about this when you run the script. If you're using regular directories, these are created, but content must be moved or copied manually before finalizing the upgrade. If you arn't using the `apps-external` directory, you must manually take care of copying only those apps which are not part of the new source.
```

When the script has finished, continue with the xref:upgrade[Upgrade] step described below.

> When the upgrade has finished, you can re-run this script to secure the `.htaccess` files.

Reviewer: ```suggestion
After the upgrade is finished, you can re-run this script to secure the `.htaccess` files.
```

=== Copy Old Apps

If you are using third party or enterprise applications, look in your new `/var/www/owncloud/apps/` or `/var/www/owncloud/apps-external/` directory to see if they are present. If not, copy them from your old instance to your new one.

NOTE: Make sure that all app directories that are defined in the `apps_paths` section of your `config.php` file do exist in your new `/var/www/owncloud/` directory.

=== Permissions

To finalize the preparation of the upgrade, you need to set the correct ownership and permissions of the new ownCloud files and folders.

.Set correct ownership
[source,console]
----
sudo chown -R www-data:www-data /var/www/owncloud
----

.Set correct permissions

> Use `chmod` depending on files and directories with different permissions:

Reviewer: ```suggestion
Use `chmod` on files and directories with different permissions:
```

The upgrade operation can take anywhere from a few minutes to a few hours, depending on the size of your installation.
When it is finished you will see either a success message or an error message that indicates why the process did not complete successfully.

=== Strong Permissions for .htaccess

.Set strong permissions for the .haccess files

> * Use `chmod` with `0640` for the .htaccess files

Reviewer: ```suggestion
* Use `chmod` with `0640` for the .htaccess files.
```

sudo service apache2 start
----

=== Check the Upgrade

> With maintenance mode disabled and the web server running, login via the web interface and perform the following steps:

Reviewer: ```suggestion
With maintenance mode disabled and the web server running, log in via the web interface and perform the following steps:
```
